### ISSUE TITLE
Force-per-voxel or Force per m^3 in FEA?

### ISSUE THREAD

**Issue by Loiisso** [dl4to/dl4to#12]
Hello!

Thank you for the package, interface between physical objects and DL is a wonderful playground!

To the issue: I am confused on the "Force" interpretation across the documentation and code-base. In tutorial [here] (https://dl4to.github.io/dl4to/tutorial_problems_and_solutions.html): the problem is formulated as such:
```
# scalar material properties
E = 7e10 # Young's modulus (in Pascal)
ν = .3 # Poisson's ratio
σ_ys = 4.5e8 # Yield stress (in Pascal)

shape = [40, 4, 8] # number of voxels in x, y and z direction
h = [0.0250, 0.0250, 0.0250] # spacial dimensions of each voxel in x, y and direction (in meters)

# define external forces
F = torch.zeros(3, *shape)
F[-1, :, :, -1] = -4e7 # external forces (in Newton)

# define locations of homogeneous Dirichlet conditions
Ω_dirichlet = torch.zeros(3, *shape)
Ω_dirichlet[:, 0, :, :] = 1

# define design space
Ω_design = -torch.ones(1, *shape) # -1s indicate that these voxels can be freely optimized
Ω_design[:,:2,:,:] = 1. # we set densities to 1 where there are Dirichlet boundary conditions
Ω_design[:,:,:,-2:] = 1. # we set densities to 1 where there are external forces
```

with note above stating that: 

> A (3 x n_x x n_y x n_z)-tensor called **_F_** which encodes external values given in **N/m3**

**Question:** where is voxel size or force normalisation accounted for in the PDE/Problem definition? Force-per-voxel looks like the interpretation based on the rest of the FEA/PDE code, but force-per-m3 looks looks far more realistic based on the FEA simulation results. The only normalisation I could find is by Youngs' Modulus, but I don't think it's related to volume normalisation.

(Force-per-voxel interpretation  load would be catastrophic for the tutorial: `F[-1, :, :, -1]=-4e7` is supposed to load "top plate" in "downward direction".  4e7 Newtons per voxel * 160 voxels -> top plate of (100cm x 10cm) is loaded with  6.4e9 / 9.81 ≈ 6.525e8 kg (≈ 652,500 tonnes). FEA analysis from the same tutorial shows stress below breaking point too)

**Comment by Loiisso** [comment]
To partially answer my own question (but would still be interesting to hear from the authors):
Likely the "size" of the voxel is accounted in the assembly of the stiffness matrix (**A**) in the PDE code, in a way that allows "F" to be treated as "force density" (F/m^3) in the right side of the "**Au=F**" situation. As we only care about finding "displacements" (u) of individual voxels for all the next steps. (There is also normalisation by Young's modulus, but not relevant here)

Specifically
```
@staticmethod
def du_dx_forward(u, h):
    du = torch.zeros_like(u)
    du[:, 0:-1,:,:] = (u[:,  1:,:,:] - u[:, 0:-1,:,:]) / h[0]
    du[:, -1  ,:,:] = (u[:, -1 ,:,:] - u[:, -2  ,:,:]) / h[0]
    return du
```
are used to assemble the 
```
def _J(self, u, dirichlet=False):
    J = lambda u: torch.cat([
        FDMDerivatives.du_dx(u, self.h, self.use_forward_differences),
        FDMDerivatives.du_dy(u, self.h, self.use_forward_differences),
        FDMDerivatives.du_dz(u, self.h, self.use_forward_differences)
    ], dim=0)
    if dirichlet:
        return FDMAssembly.apply_dirichlet_zero_columns_to_operator(J, self.Ω_dirichlet)(u)
    return J(u)
```

which is part of the "forward" calculation.
